src/callback/file_callbacks.h

```diff
@@ -7,6 +7,7 @@
 #include "../helpers/containers.h"
 #include <thread>
 #include <condition_variable>
+#include <cstring> // This include is required for both Linux and MacOS targets as they don't include the necessary headers for 'memcpy' by default
```

Comment:
👀

Reply:
Classical Linux/MacOS C++ thing 🤷 
I left a comment such that it is clear that this is not required for Windows.

Reply:
I think it is good as it is.